### ISSUE TITLE
Small mAP in train and val

### ISSUE THREAD

**Issue by filetii** [giuliomattolin/ConfMix#17]
I've been working with a few different datasets using your model, including the Sim10k --> Cityscapes experiment, but just as @AmrimKareen, I'm getting tiny mAP results from my runs. I also checked the other issue posted with the same problem in which you suggested checking if all the files are in the correct path in the .yaml and I'm pretty sure it is, since the train.py output shows: 

"[34m[1mtrain: [0mScanning '/home/dvmfileti@pos.usricmc.icmc.usp.br/IC/modelos_ic/confmix_experiments/ConfMix/datasets/Sim10k/labels/train.cache' images and labels... 10000 found, 0 missing, 25 empty, 0 corrupt: 100%|██████████| 10000/10000 [00:00<?, ?it/s]
[34m[1mtrain: [0mScanning '/home/dvmfileti@pos.usricmc.icmc.usp.br/IC/modelos_ic/confmix_experiments/ConfMix/datasets/Sim10k/labels/train.cache' images and labels... 10000 found, 0 missing, 25 empty, 0 corrupt: 100%|██████████| 10000/10000 [00:00<?, ?it/s]

[34m[1mval: [0mScanning '/home/dvmfileti@pos.usricmc.icmc.usp.br/IC/modelos_ic/confmix_experiments/ConfMix/datasets/Cityscapes/adv_KITTI/labels/val/frankfurt.cache' images and labels... 475 found, 25 missing, 0 empty, 0 corrupt: 100%|██████████| 500/500 [00:00<?, ?it/s]
[34m[1mval: [0mScanning '/home/dvmfileti@pos.usricmc.icmc.usp.br/IC/modelos_ic/confmix_experiments/ConfMix/datasets/Cityscapes/adv_KITTI/labels/val/frankfurt.cache' images and labels... 475 found, 25 missing, 0 empty, 0 corrupt: 100%|██████████| 500/500 [00:00<?, ?it/s]
Plotting labels to runs/train/sim10k_sem_UDA2/labels.jpg... "

So indeed the model found the images and labels. Any ideas on how to solve this?

**Comment by giuliomattolin** [comment]
Hi @filetii,

That's strange, the paths indeed seem to be correct from the output snippet you posted. Try to delete the `.cache` files in the labels' directories and re-run the training, maybe they contain some old and incorrect information. Moreover, what phase of the training are you running? Note that the final model is obtained after two phases: phase 1 (model pre-train) and phase 2 (adaptive learning), whose running scripts are both illustrated in the README.

**Comment by filetii** [comment]
Hi @giuliomattolin ! Thank you for your answer!

As for the training, I am aware of the two-phase strategy, and as such, tested the model in the checkpoint of the second one! However I noticed that the tiny mAP (around 0.01 to 0.0003) are from a single dataset. Im also adapting sim10k to cityscapes and i got pretty decent mAP (0.5 and above! I hope the mAP are in decimals and in your article for a better visualization you multiplied it right?) It's a single dataset that proves a bad performance...

As such I think it might be possible and reasonable that the dataset might be harder to adapt or something like that right?  It's an italian gallery of art, a synthetic and a real gopro dataset. The other ones showed nice results, so it's more likely that the problem is in my dataset rather than your models or my training! 

I will check out on this later! And since you are reading my answer, I would like to let you know that your work will probably be part of my projects here at USP in Brazil, and I do expect to publish an article regarding it and some other domain adaptation methods next year, and of course I will cite you there! Congrats on the great work :)

**Comment by giuliomattolin** [comment]
Ok, got it! Outside of the benchmarks we tested the method, unfortunately I can't help you much since I don't have direct experience. Maybe it requires some hyperparameter tuning, e.g. an higher confidence threshold for filtering pseudo detections could help create fewer false positives and generate more reliable labels for the mixed samples. That's my two cents on improving the performance of your dataset. Thank you for your interest and I'm looking forward to reading your article!
